lib/march_hare/session.rb

```diff
@@ -110,6 +117,8 @@ def self.connect(options = {})
           ensure
             is.close if is
           end
+        elsif options[:trust_manager]
+          cf.use_ssl_protocol(tls, options[:trust_manager])
         else
           cf.use_ssl_protocol(tls)
```

Comment:
We should remove these options as part of this patch IMHO. These are unsafe https://github.com/rabbitmq/rabbitmq-java-client/blob/5e0414a87778969aa919172685e0095343980198/src/main/java/com/rabbitmq/client/ConnectionFactory.java#L595

Reply:
It is as safe as the trust manager the user provides is.

Reply:
That may be factually true, but there's nothing about this that is obvious. These options should have a big warning on them, something like `use_unsafe_testing_ssl`.

SSL without peer verification is normal. I realize the RabbitMQ java client does this, but it encourages bad behavior.

Reply:
I don’t disagree with this. There is nothing wrong with the option, however. Having a default that enables peer certificate chain verification is a good idea but… it’d be a major breaking change and a ton of support load because developers don’t usually know much about how TLS or peer verification actually work. That’s the only reason why we decided to keep this default in the Java client.

Bunny and Erlang clients log a warning when peer verification. I don’t remember if Java and .NET ones do but if not, we should make them. That’s the best option for March Hare as well.

Reply:
Thanks for the considered response @michaelklishin 

I think this discussion shouldn't block this ticket. I'll open a new one.